pyanaconda/modules/storage/bootloader/systemd.py

```diff
@@ -0,0 +1,190 @@
+    _config_dir = "efi/loader/"
+    stage2_max_end = None
+
+    stage2_is_valid_stage1 = True
+    stage2_bootable = True
+    stage2_must_be_primary = False
+
+    # requirements for boot devices
+    stage2_device_types = ["partition"]
+
+    def __init__(self):
+        super().__init__()
+        self.encrypted_password = ""
+
+    #
+    # configuration
+    #
+
+    @property
+    def config_dir(self):
+        """ Full path to configuration directory. """
+        log.info("systemd.py: config_dir")
+        return "/boot/" + self._config_dir
+
+    @property
+    def config_file(self):
+        """ Full path to configuration file. """
+        log.info("systemd.py: config_file")
+        return "%s/%s" % (self.config_dir, self._config_file)
+
+    @property
+    def stage2_format_types(self):
+        log.info("systemd.py: stage2_format_types")
+        if productName.startswith("Red Hat "): # pylint: disable=no-member
+            return ["xfs", "ext4", "ext3", "ext2"]
+        else:
+            return ["ext4", "ext3", "ext2", "btrfs", "xfs"]
+
+
+    def write_device_map(self):
+        """Write out a device map containing all supported devices."""
+        log.info("systemd.py: write_device_map")
+        map_path = os.path.normpath(conf.target.system_root + self.device_map_file)
+        if os.access(map_path, os.R_OK):
+            os.rename(map_path, map_path + ".anacbak")
+
+        devices = self.disks
+        if self.stage1_device not in devices:
+            devices.append(self.stage1_device)
+
+        for disk in self.stage2_device.disks:
+            if disk not in devices:
+                devices.append(disk)
+
+        devices = [d for d in devices if d.is_disk]
+
+        if len(devices) == 0:
+            return
+
+    # copy console update from grub2.py
+    def write_config_console(self, config):
+        log.info("systemd.py: write_config_console")
+        if not self.console:
+            return
+
+        console_arg = "console=%s" % self.console
+        if self.console_options:
+            console_arg += ",%s" % self.console_options
+        self.boot_args.add(console_arg)
+
+
+
+    def write_config(self):
+        log.info("systemd.py: write_config systemd start")
+        # So at this point, all this is pretty much redundant
+        self.write_config_console(None)
+
+        # just rewrite the loader.conf
+        #cfg_file = config_file(self)
+        config_path = "%s%s" % (conf.target.system_root, "/boot/efi/loader/loader.conf")
+        log.info("systemd.py: write_config systemd loader conf : %s ", config_path)
+
+        # outside of resetting the default, this file should probably be moved to
+        # stubby as well?
+        config = open(config_path, "w")
+        config.write("timeout 3\n")
+        # create a default entry? No need right now..
+        # config.write("default 4142287a09f64398a50ac387bf9e1fca-*\n")
+        config.write("#console-mode keep\n")
+        config.close()
+
```

Comment:
I wonder if all this should be done manually or could just `bootctl install` be invoked ?

Reply:
Again IIRC, the default loader.conf has the timeout parm commented out, which rather than defaulting to three actually means that it defaults to 0.

Reply:
I double-checked this, and yes, I think, for the moment, we want to rewrite it to enable a non-zero timeout.

Reply:
I'm going to close this one out, mainly because the current code is honoring the anaconda bootloader timeout parameter and passing it into systemd-boot via this config file. Which is IMHO definitely the right thing to be doing and requires modifying the loader.conf file. Whether that is done as writing a new/clean copy or trying to sed/update the one placed there by bootctl I guess could be a discussion, but AFAIU, that wasn't really what your point here was.